src/holdings/holdings.py

```diff
@@ -38,6 +39,31 @@ def add_holding(self, security_id: UUID, holding_date: date, amount: Decimal):
 
         self.data[security_id][holding_date] += amount
 
+    def calculate_amounts(self, start_date: date, end_date: date):
+        """
+        Calculates daily amounts for holdings. This method needs to be called before using get_day_sum method
+
+        Args:
+            start_date: start date for the date range
+            end_date: end date for the date range
+        """
+        for security_id in self.get_security_ids():
+            self.day_amounts[security_id] = {}
+            security_min_date = self.get_security_min_date(security_id)
```

Comment:
change the get_security_min_date to( same with max method):

def get_security_min_date(self, security_id: UUID) -> Optional[date]:

        result = None

        if security_id not in self.data:
            return None

        security_holdings = self.data[security_id]
        return min(security_holdings.keys())